core/store/models/run_output.go

```diff
@@ -0,0 +1,119 @@
+package models
+
+import (
+	"fmt"
+
+	"github.com/tidwall/gjson"
+)
+
+// RunOutput represents the result of performing a Task
+type RunOutput struct {
+	data   JSON
+	status RunStatus
+	err    error
+}
+
+// NewRunOutputError returns a new RunOutput with an error
+func NewRunOutputError(err error) RunOutput {
+	return RunOutput{
+		status: RunStatusErrored,
+		err:    err,
+	}
+}
+
+// NewRunOutputCompleteWithResult returns a new RunOutput that is complete and
+// contains a result
+func NewRunOutputCompleteWithResult(resultVal interface{}) RunOutput {
+	var data JSON
+	data, _ = data.Add("result", resultVal)
```

Comment:
Is there a reason the error from JSON.Add(..) is ignored?  It looks like the function could return error if an attempt to marshal resultVal fails, which this function does not have control over.

Reply:
Agreed. Seems like this should result in returning `NewRunOutputError` instead.

Reply:
This can’t actually error! But I don’t think that will be satisfying for anyone, so I will rewrite this using an infallible type.